androidbrowserhelper/src/main/java/com/google/androidbrowserhelper/trusted/DelegationService.java

```diff
@@ -14,19 +14,145 @@
 
 package com.google.androidbrowserhelper.trusted;
 
+import android.Manifest;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.os.Bundle;
+import android.os.Handler;
+import android.os.Looper;
+import android.os.Messenger;
+import android.os.RemoteException;
+import android.text.TextUtils;
+import android.util.Log;
+
 import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
 import androidx.browser.trusted.TokenStore;
+import androidx.browser.trusted.TrustedWebActivityCallbackRemote;
+import androidx.core.content.ContextCompat;
 
 /**
  * An extension of {@link androidx.browser.trusted.TrustedWebActivityService} that implements
  * {@link androidx.browser.trusted.TrustedWebActivityService#getTokenStore()} using a
  * {@link SharedPreferencesTokenStore}.
  */
 public class DelegationService extends androidx.browser.trusted.TrustedWebActivityService {
+    private static final String TAG = "TWADelegationService";
+
+    private static final String CHECK_LOCATION_PERMISSION_COMMAND_NAME = "checkAndroidLocationPermission";
+    private static final String LOCATION_PERMISSION_RESULT = "locationPermissionResult";
+
+    private static final String START_LOCATION_COMMAND_NAME = "startLocation";
+    private static final String STOP_LOCATION_COMMAND_NAME = "stopLocation";
+
+    private static final String EXTRA_RESULT_RECEIVER = "EXTRA_RESULT_RECEIVER";
+    private static final String EXTRA_PERMISSIONS = "EXTRA_PERMISSIONS";
+    private static final String EXTRA_GRANT_RESULTS = "EXTRA_GRANTED_RESULT";
+
+    private static final String mLocationPermission = Manifest.permission.ACCESS_FINE_LOCATION;
+
+    private boolean mIsProviderGmsCore;
 
     @NonNull
     @Override
     public TokenStore getTokenStore() {
         return new SharedPreferencesTokenStore(this);
     }
+
+    @Nullable
+    @Override
+    public Bundle onExtraCommand(
+            String commandName, Bundle args, TrustedWebActivityCallbackRemote callback) {
+        Log.d(TAG, commandName);
+        Bundle result = new Bundle();
+        switch (commandName) {
+            case CHECK_LOCATION_PERMISSION_COMMAND_NAME: {
+                if (callback == null) break;
+
+                result.putBoolean(commandName, true);
```

Comment:
(I mentioned this to you in an email, here's probably a better place for it.)

How come we're using the commandName as a key for the success value? Couldn't we just use "success"?